sycl/include/CL/sycl/ONEAPI/intel_matrix/matrix-amx.hpp

```diff
@@ -0,0 +1,439 @@
+} // namespace ext
+} // namespace sycl
+
+namespace sycl {
+namespace ext::intel {
+namespace matrix {
+using namespace cl::sycl;
+using namespace cl::sycl::ONEAPI;
+
+#ifdef __SYCL_DEVICE_ONLY__
+SYCL_EXTERNAL extern "C" _tile1024i
+_tileloadd64_internal(short row, short col, char *buf, size_t stride);
+SYCL_EXTERNAL extern "C" _tile1024i
+_tdpbssd_internal(unsigned short m, unsigned short n, unsigned short k,
+                  _tile1024i dst, _tile1024i src1, _tile1024i src2);
+SYCL_EXTERNAL extern "C" _tile1024i
+_tdpbf16ps_internal(unsigned short m, unsigned short n, unsigned short k,
+                    _tile1024i dst, _tile1024i src1, _tile1024i src2);
+SYCL_EXTERNAL extern "C" void _tilestored64_internal(short row, short col,
+                                                     char *buf, size_t stride,
+                                                     _tile1024i tile);
+static _tile1024i tileloadd64_internal(short row, short col, char *buf,
+                                       size_t stride) {
+  return _tileloadd64_internal(row, col, buf, stride);
+}
+static _tile1024i tdpbssd_internal(unsigned short m, unsigned short n,
+                                   unsigned short k, _tile1024i dst,
+                                   _tile1024i src1, _tile1024i src2) {
+  return _tdpbssd_internal(m, n, k, dst, src1, src2);
+}
+static _tile1024i tdpbf16ps_internal(unsigned short m, unsigned short n,
+                                     unsigned short k, _tile1024i dst,
+                                     _tile1024i src1, _tile1024i src2) {
+  return _tdpbf16ps_internal(m, n, k, dst, src1, src2);
+}
+static void tilestored64_internal(short row, short col, char *buf,
+                                  size_t stride, _tile1024i tile) {
+  return _tilestored64_internal(row, col, buf, stride, tile);
+}
+#endif
+
+#ifndef __SYCL_DEVICE_ONLY__
+static _tile1024i tileloadd64_internal(short row, short col, char *buf,
+                                       size_t stride) {
+  return __builtin_ia32_tileloadd64_internal(row, col, buf, stride);
+}
+static _tile1024i tdpbssd_internal(unsigned short m, unsigned short n,
+                                   unsigned short k, _tile1024i dst,
+                                   _tile1024i src1, _tile1024i src2) {
+  return __builtin_ia32_tdpbssd_internal(m, n, k, dst, src1, src2);
+}
+static _tile1024i tdpbf16ps_internal(unsigned short m, unsigned short n,
+                                     unsigned short k, _tile1024i dst,
+                                     _tile1024i src1, _tile1024i src2) {
+  return __builtin_ia32_tdpbf16ps_internal(m, n, k, dst, src1, src2);
+}
+static void tilestored64_internal(short row, short col, char *buf,
+                                  size_t stride, _tile1024i tile) {
+  __builtin_ia32_tilestored64_internal(row, col, buf, stride, tile);
+}
+#endif
+
+enum class matrix_layout { row_major, col_major, packed_a, packed_b };
+
+#define TILE_SZ 16
```

Comment:
```suggestion
inline constexpr size_t tile_size = 16;
```
`TILE_SZ` is not undefined, so it leaks into user's code and might conflict with user's code. Using constexpr is much better way to define a constant.